sympy/polys/laurent.py

```diff
@@ -0,0 +1,566 @@
+from __future__ import annotations
+
+from functools import reduce
+from operator import add, mul
+
+from sympy.core.expr import Expr
+from sympy.core.add import Add
+from sympy.core.symbol import Symbol
+from sympy.core.sympify import sympify, CantSympify
+from sympy.polys.domains.domainelement import DomainElement
+from sympy.polys.domains.laurentpolynomialring import LaurentPolynomialRing
+from sympy.polys.fields import FracField
+from sympy.polys.orderings import LexOrder
+from sympy.polys.polyerrors import CoercionFailed
+from sympy.polys.rings import PolyElement, PolyRing
+from sympy.printing.defaults import DefaultPrinting
+
+from typing import Any, Optional
+
+
+def laurent_ring(symbols, domain, order='lex'):
+    """Construct a Laurent polynomial ring.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.laurent import laurent_ring
+    >>> from sympy import ZZ
+    >>> R, x, y = laurent_ring('x,y', ZZ, order='lex')
+    >>> R
+    Laurent polynomial ring in x, y over ZZ with lex order
+    >>> p = x**2 + 1/y
+    >>> p
+    x**2 + 1/y
+    >>> p ** 2
+    x**4 + 2*x**2/y + 1/y**2
+    """
+    ring = LaurentPolyRing(symbols, domain, order=order)
+    return (ring,) + ring.gens
+
+
+class LaurentPolyRing:
+    """Ring of Laurent polynomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.laurent import LaurentPolyRing
+    >>> from sympy import ZZ
+    >>> from sympy.abc import x, y
+    >>> R = LaurentPolyRing([x, y], ZZ, order='lex')
+    >>> R
+    Laurent polynomial ring in x, y over ZZ with lex order
+    >>> p = R.from_expr(1/x + x*y)
```

Comment:
Should this line be deleted? Or have a `>>> p` line after it, showing the result?

Reply:
Thanks.

It should be deleted. I've deleted it now.